crypto/x509/x509_lu.c

```diff
@@ -323,33 +328,47 @@ int X509_STORE_CTX_get_by_subject(const X509_STORE_CTX *vs,
 
     if (!X509_STORE_lock(store))
         return 0;
-
     tmp = X509_OBJECT_retrieve_by_subject(store->objs, type, name);
     X509_STORE_unlock(store);
 
     if (tmp == NULL || type == X509_LU_CRL) {
         for (i = 0; i < sk_X509_LOOKUP_num(store->get_cert_methods); i++) {
             lu = sk_X509_LOOKUP_value(store->get_cert_methods, i);
-            j = X509_LOOKUP_by_subject_ex(lu, type, name, &stmp, vs->libctx,
-                                          vs->propq);
-            if (j) {
+            if (lu->skip)
+                continue;
+            if (lu->method == NULL)
+                return -1;
+            j = X509_LOOKUP_by_subject_ex(lu, type, name, &stmp,
+                                          vs->libctx, vs->propq);
+            if (j != 0) { /* non-zero value is considered success here */
                 tmp = &stmp;
                 break;
             }
         }
         if (tmp == NULL)
             return 0;
     }
+    if (tmp->type != type)
```

Comment:
This seems like a strange check here... Is that even possible that the type doesnt match

Reply:
The check I inserted here was just for being more defensive,
but you are right that by construction of ` X509_OBJECT_retrieve_by_subject()` this cannot really go wrong,
so I've removed the check again.